remotes/docker/fetcher.go

```diff
@@ -71,8 +71,16 @@ func (r dockerFetcher) Fetch(ctx context.Context, desc ocispec.Descriptor) (io.R
 				Host:         u.Host,
 				Scheme:       u.Scheme,
 				Path:         u.Path,
-				Capabilities: HostCapabilityPull,
+				Capabilities: HostCapabilityExternalObject,
 			}
+			// add the Authorizer for the first host that has the external object
+			// capability as the Authorizer for the RegistryHost
+			for _, h := range r.filterHosts(HostCapabilityExternalObject) {
+				if h.Authorizer != nil {
```

Comment:
This should likely match against at least the hostname as well. Possibly scheme and path prefix. Header and Client could also be overriden.

Reply:
Updated to match the hostname and scheme.
Didn't do path prefix because if the manifest is something like

```json
"layers": [
    {
      "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
      "digest": "sha256:25da54cc0a08f4ca602c6bcd3e52d70082eb8a25ee022bc9f1dda019de49197a",
      "size": 123,
      "urls": [
        "https://path.to.external.url/blobs/123456789"
      ]
    }
  ]
```

the path after doing the url parsing will be `blobs/123456789` which varies per blob

Updated to override the `Header` and `Client` on a match of the HostRegistry.